### ISSUE TITLE
API Documentation wanted.

### ISSUE THREAD

**Issue by netflash** [ustream/openduty#43]
Hi,
could you please help me a little bit with how to start working with API?
I was striving to use this as an example https://developer.pagerduty.com/documentation/rest/services/list
```
curl -v http://localhost/api/v1/services
```
but no luck.

I understand that I missing something, just can't understand what :-)

**Comment by deathowl** [comment]
Hi! No sadly, you dont miss a thing, Services API is not done yet, but I'll do it.

**Comment by theshades** [comment]
Any update on this? If need help I may be able to provide some python or perl help.

**Comment by deathowl** [comment]
Any help is welcome as at the moment I'm a bit short on time :)

**Comment by sofixa** [comment]
Just FYI, the company i work at is interested in OpenDuty and we've done some dev to make OpenDuty more usable for our use cases(like various template fixes, finer ACLs, more notification methods, etc.). In a few weeks i plan to submit a PR with all our changes(or at least those that aren't too specific to us). That should include some documentation as well(as other people have noticed, the code base is pretty good, but it isn't very well documented, and that severely hampers usability and extensibility).

I don't imagine it'll take a lot of time to make a services API, and, in any case, we need it as well, so i might look into it, but it won't be for at least another week.

Regards,
Adrian

**Comment by deathowl** [comment]
Good stuff guys, I'm more than glad to welcome your contributions, and do my part in reviewing and merging them if so.

**Comment by deathowl** [comment]
@sofixa any update on this?

**Comment by sofixa** [comment]
@deathowl mm kinda. 

We have added a lot of stuff(groups, unacknowledge, resolve notifications, some UI stuff(icons, confirmation before deletion), and a few notification methods(HipChat, sms-tools) come to mind, there's more i can't think of now), and we intend to publish everything, but we haven't had the time to properly check, sanitize and document everything so its open sourcable :/ We'll discuss this next week again and i hope we'll find the time. 

Besides that, we've been running it in production for a few months now and we're pretty happy.

**Comment by andr01droth** [comment]
@sofixa do you still plan on publishing your enhancements?

**Comment by sofixa** [comment]
@andr01droth Yes, but sadly i can't provide an ETA. I'll try to take a look this weekend but can't promise anything.

**Comment by allahbaksh** [comment]
@sofixa Can you please squeeze some time if possible for this.

**Comment by sujen** [comment]
@sofixa If you could share the details/PR/documentation you have worked out so far, that would be helpful.. thaks
